File: docs/snippets/all/tutorials/getting_started_log.cpp
#include <dalaran.hpp>

#include <cmath>

int main(int argc, char* argv[]) {
    const auto rec =
        dalaran::RecordingStream("dalaran_example_getting_started", "run-1");
    rec.save("run-1.rrd").exit_on_failure();

    for (int t = 0; t < 10; ++t) {
        rec.set_time_sequence("step", t);
        const auto tf = static_cast<double>(t);
        rec.log("/arm/shoulder", dalaran::Scalars(std::sin(tf * 0.5)));
        rec.log("/arm/elbow", dalaran::Scalars(std::cos(tf * 0.5)));
    }
}

File: docs/snippets/all/tutorials/getting_started_log.rs
fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec =
        dalaran::RecordingStreamBuilder::new("dalaran_example_getting_started")
            .recording_id("run-1")
            .save("run-1.rrd")?;

    for t in 0..10 {
        rec.set_time_sequence("step", t);
        let tf = t as f64;
        rec.log("/arm/shoulder", &dalaran::Scalars::single((tf * 0.5).sin()))?;
        rec.log("/arm/elbow", &dalaran::Scalars::single((tf * 0.5).cos()))?;
    }

    Ok(())
}
